### PR TITLE
MAGETWO-84414 indexer DefaultStock

### PR DESCRIPTION
**Pull request by m-milewski**
### Description
Deleting old records before each update index table.

### Fixed Issues (if relevant)
1. #84414: Stock inventory reindex bug

### Manual testing scenarios
1. Mass update over 1000 products.

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=14139) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=14139) before we can accept your contribution.<br/>

**Comment by rogyar** [comment]
Hello @m-milewski. Please, consider fixing the failing integration tests. Thank you

**Comment by m-milewski** [comment]
@rogyar I fixed fails in integration tests

File: app/code/Magento/CatalogInventory/Model/ResourceModel/Indexer/Stock/DefaultStock.php [hunk 1]
```diff
@@ -305,14 +305,16 @@ protected function _updateIndex($entityIds)
...
+                $this->deleteOldRecords(array_column($data, 'product_id'));
                 $this->_updateIndexTable($data);
                 $data = [];
             }
         }
 
-        $this->deleteOldRecords($entityIds);
-        $this->_updateIndexTable($data);
-
+        if (is_array($data) && !empty($data)) {
```
Comment (rogyar) [inline app/code/Magento/CatalogInventory/Model/ResourceModel/Indexer/Stock/DefaultStock.php:314]: If we are considering fix for the following issue https://github.com/magento/magento2/issues/12205, it's already fixed withing a scope of 2.2-develop. It should be enough to remove all records at the very beginning of the index update operation without performing a separate delete in a loop. 

Please, check the latest commit here https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/CatalogInventory/Model/ResourceModel/Indexer/Stock/DefaultStock.php for the clarification.

Thank you

**Review by rogyar — CHANGES_REQUESTED** [review]

**Comment by sidolov** [comment]
@m-milewski , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for collaboration!

**Pull request closed without merging**